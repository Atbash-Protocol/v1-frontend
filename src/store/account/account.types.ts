import { StaticJsonRpcProvider, JsonRpcProvider } from '@ethersproject/providers';
import { Networks } from 'constants/index';
import { Bond } from 'helpers/bond/bond';
import { IToken } from 'helpers/tokens';

export interface IGetBalances {
    address: string;
    networkID: Networks;
    provider: StaticJsonRpcProvider | JsonRpcProvider;
}

export interface IAccountBalances {
    balances: {
        wsBASH: string;
        sBASH: string;
        BASH: string;
        aBash: string;
    };
}

export interface ILoadAccountDetails {
    address: string;
    networkID: Networks;
    provider: StaticJsonRpcProvider | JsonRpcProvider;
}

export interface IUserAccountDetails {
    balances: {
        BASH: string;
        sBASH: string;
        wsBASH: string;
        aBash: string;
    };
    redeeming: {
        BASH: number;
        aBash: number;
    };
    staking: {
        BASH: number;
        sBASH: number;
    };
    wrapping: {
        sBASHAllowance: number;
    };
}

export interface ICalcUserBondDetails {
    address: string;
    bond: Bond;
    provider: StaticJsonRpcProvider | JsonRpcProvider;
    networkID: Networks;
}

export interface ICalcUserTokenDetails {
    address: string;
    token: IToken;
    provider: StaticJsonRpcProvider | JsonRpcProvider;
    networkID: Networks;
}

export interface IAccountSlice {
    bonds: { [key: string]: IUserBondDetails };
    balances: {
        wsBASH: string;
        sBASH: string;
        BASH: string;
    };
    loading: boolean;
    redeeming: {
        BASH: number;
    };
    staking: {
        BASH: number;
        sBASH: number;
    };
    wrapping: {
        sBASHAllowance: number;
    };
    tokens: { [key: string]: IUserTokenDetails };
}

export interface IUserTokenDetails {
    allowance: number;
    balance: number;
    isAvax?: boolean;
}

export interface IUserBondDetails {
    allowance: number;
    balance: number;
    avaxBalance: number;
    interestDue: number;
    bondMaturationBlock: number;
    pendingPayout: number; //Payout formatted in gwei.
}
