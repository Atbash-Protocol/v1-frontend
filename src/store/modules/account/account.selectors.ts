import Decimal from 'decimal.js';
import { utils } from 'ethers';
import { createSelector } from 'reselect';

import { RootState } from 'store/store';

export const selectBASHBalance = (state: RootState): Decimal => {
    const BASHAmount = state.account.balances.BASH; // 9 Decimals

    return new Decimal(BASHAmount.toString()).div(10 ** 9);
};

export const selectSBASHBalance = (state: RootState): Decimal => {
    const SBASHAmount = state.account.balances.SBASH; // 9 Decimals

    return new Decimal(SBASHAmount.toString()).div(10 ** 9);
};

export const selectWSBASHBalance = (state: RootState): Decimal => {
    const WSBASHAmount = state.account.balances.WSBASH;

    return new Decimal(WSBASHAmount.toString()).div(10 ** 9); // 9 Decimals
};

export const selectFormattedStakeBalance = createSelector([selectBASHBalance, selectSBASHBalance, selectWSBASHBalance], (BASHBalance, SBASHBalance, WSBASHBalance) => {
    return Object.entries({ BASH: BASHBalance, SBASH: SBASHBalance, WSBASH: WSBASHBalance }).reduce(
        (acc, [key, amount]) => {
            return {
                ...acc,
                [key]: [amount.toFixed(2), key.toLocaleUpperCase()].join(' '),
            };
        },
        {
            BASH: '0.00 BASH',
            SBASH: '0.00 SBASH',
            WSBASH: '0.00 WSBASH',
        },
    );
});

export const selectFormattedBASHBalance = (state: RootState): string => {
    const BASHAmount = state.account.balances.BASH; // 9 Decimals

    return [Number(utils.formatUnits(BASHAmount, 'gwei')).toFixed(2), 'BASH'].join(' ');
};

export const selectUserStakingAllowance = (state: RootState) => {
    const { BASH, SBASH } = state.account.stakingAllowance;

    return {
        BASHAllowanceNeeded: BASH.eq(0),
        SBASHAllowanceNeeded: SBASH.eq(0),
    };
};

export const selectAccountLoading = (state: RootState): boolean => state.account.loading;
