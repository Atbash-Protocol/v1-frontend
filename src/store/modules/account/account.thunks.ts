import { JsonRpcProvider } from "@ethersproject/providers";
import { createAsyncThunk } from "@reduxjs/toolkit";
import { MemoTokenContract, TimeTokenContract } from "abi";
import { getAddresses } from "constants/addresses";
import { ethers } from "ethers";
import { IReduxState } from "store/slices/state.interface";

export const loadBalancesAndAllowances = createAsyncThunk("account/loadBalancesAndAllowances", async ({ chainID, provider, address }: any, { getState }): Promise<any> => {
    let BASHbalance = 0;
    let sBASHBalance = 0;
    let wsBASHBalance = 0;
    let stakeAllowance = 0;
    let unstakeAllowance = 0;
    let wrapAllowance = 0;
    let redeemAllowance = 0;

    const {
        main: {
            contracts: { BASH_CONTRACT, SBASH_CONTRACT, STAKING_ADDRESS, STAKING_HELPER_ADDRESS, WSBASH_ADDRESS },
        },
    } = getState() as IReduxState;

    if (BASH_CONTRACT) {
        BASHbalance = await BASH_CONTRACT.balanceOf(address);
        stakeAllowance = await BASH_CONTRACT.allowance(address, STAKING_HELPER_ADDRESS?.address);
        // disable: redeemAllowance = await sbContract.allowance(address, addresses.REDEEM_ADDRESS);
    }

    if (SBASH_CONTRACT) {
        try {
            wrapAllowance = await SBASH_CONTRACT.allowance(address, WSBASH_ADDRESS?.address);
            unstakeAllowance = await SBASH_CONTRACT.allowance(address, STAKING_ADDRESS?.address);
        } catch (err) {
            console.log(err);
        }
    }

    // if (addresses.WSBASH_ADDRESS) {
    //     const wsBASHContract = new ethers.Contract(addresses.WSBASH_ADDRESS, SBashTokenContract, provider);
    //     wsBASHBalance = await wsBASHContract.balanceOf(address);
    // }

    return {
        balances: {
            WSBASH: ethers.utils.formatEther(wsBASHBalance),
            SBASH: ethers.utils.formatUnits(sBASHBalance, "gwei"),
            BASH: ethers.utils.formatUnits(BASHbalance, "gwei"),
        },

        stakingAllowance: {
            BASH: Number(stakeAllowance),
            SBASH: Number(unstakeAllowance),
        },
        // wrapping: {
        //     sBASHAllowance: Number(wrapAllowance),
        // },
    };
});
