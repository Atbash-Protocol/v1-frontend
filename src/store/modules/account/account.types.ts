export interface AccountSlice {
    stakingAllowance: {
        BASH: number;
        SBASH: number;
    };
    balances: {
        BASH: number;
        SBASH: number;
        WSBASH: number;
    };
    loading: boolean;
}
