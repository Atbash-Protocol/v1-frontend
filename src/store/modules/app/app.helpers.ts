import { Epoch } from './app.types';

export const calculateStakingRewards = (epoch: Epoch | null, circSupply: number) => {
    const stakingReward = epoch ? epoch.distribute : null; // the amount of BASH to distribute in the coming epoch
    const stakingRebase = stakingReward ? stakingReward.toNumber() / (circSupply * Math.pow(10, 9)) : null; // rewardYield rate for this epoch

    return {
        fiveDayRate: stakingRebase ? Math.pow(1 + stakingRebase, 5 * 3) - 1 : null, // 3 epoch/day
        stakingAPY: stakingRebase ? Math.pow(1 + stakingRebase, 365 * 3) - 1 * 100 : null,
        stakingReward,
        stakingRebase,
    };
};
