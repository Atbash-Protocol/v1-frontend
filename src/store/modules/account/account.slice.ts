import { createSlice } from "@reduxjs/toolkit";
import { BigNumber } from "ethers";
import { loadBalancesAndAllowances } from "./account.thunks";
import { AccountSlice } from "./account.types";

const initialState: AccountSlice = {
    balances: {
        BASH: BigNumber.from(0),
        SBASH: BigNumber.from(0),
        WSBASH: BigNumber.from(0),
    },
    stakingAllowance: {
        BASH: BigNumber.from(0),
        SBASH: BigNumber.from(0),
    },
    loading: true,
};

export const marketSlice = createSlice({
    name: "app-account-slice",
    initialState,
    reducers: {},
    extraReducers: builder => {
        builder.addCase(loadBalancesAndAllowances.fulfilled, (state, action) => {
            return {
                ...state,
                ...action.payload,
                loading: false,
            };
        });
        builder.addCase(loadBalancesAndAllowances.rejected, (state, action) => {
            return {
                ...state,
                loading: false,
            };
        });
    },
});

export default marketSlice.reducer;
