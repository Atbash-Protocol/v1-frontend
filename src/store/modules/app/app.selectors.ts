import Decimal from 'decimal.js';
import { BigNumber, utils } from 'ethers';
import { pick } from 'lodash';
import { createSelector } from 'reselect';

import { formatUSD } from 'helpers/price-units';
import { IReduxState } from 'store/slices/state.interface';
import { RootState } from 'store/store';

import { calculateStakingRewards } from './app.helpers';

export const selectFormattedReservePrice = (state: RootState): string | null => {
    const { reserves } = state.main.metrics;
    const { dai } = state.markets.markets;

    if (!reserves || !dai) return null;

    const reservePrice = Number(utils.formatUnits(reserves.toString(), 'gwei')) * dai;

    return formatUSD(reservePrice, 2);
};

export const selectReserve = (state: RootState): Decimal => new Decimal(state.main.metrics.reserves?.toString() ?? 0);
export const selectReserveLoading = (state: RootState): boolean => state.main.metrics.reserves === null;

export const selectBlockchainLoading = (state: RootState) => state.main.blockchain.loading === true;
export const selectMetricsLoading = (state: RootState) => state.main.metrics.loading === true;
export const selectStakingLoading = (state: RootState) => state.main.staking.loading === true;
export const selectContractsLoading = (state: RootState) => Object.values(state.main.contracts).some(contract => contract === null);

export const useContractLoaded = (state: RootState): boolean => {
    const { contracts } = state.main;

    return Object.values(contracts).every(contract => contract !== null);
};

export const selectAppLoading = createSelector(
    [selectReserveLoading, selectBlockchainLoading, selectMetricsLoading, selectBlockchainLoading, selectContractsLoading],
    (...conditions) => {
        return conditions.some(condition => condition === true);
    },
);

export const useNextRebase = (state: RootState): number | undefined => {
    const {
        main: {
            staking: { epoch },
        },
    } = state;

    return epoch?.endTime;
};

export const useBlockchainInfos = (state: RootState) => state.main.blockchain;

export const selectCirculatingSupply = (state: RootState) => state.main.metrics.circSupply;

export const selectUserStakingInfos = createSelector(
    [
        (state: IReduxState) => {
            return {
                circSupply: state.main.metrics.circSupply,
                ...pick(state.main.staking, ['epoch', 'index']),
                ...pick(state.account.balances, ['SBASH', 'WSBASH']),
            };
        },
    ],
    ({ index, epoch, circSupply, SBASH, WSBASH }) => {
        let stakingRebase = new Decimal(0);

        const SBASHBalance: Decimal = new Decimal(SBASH.div(10 ** 9).toHexString());
        const WBASHBalance: Decimal = new Decimal(WSBASH.div(10 ** 9).toHexString());

        if (epoch?.distribute && circSupply) stakingRebase = new Decimal(epoch.distribute.toHexString()).div(new Decimal(circSupply).mul(10 ** 9));

        const nextRewardValue = stakingRebase.mul(SBASHBalance);
        const { fiveDayRate } = calculateStakingRewards(epoch, circSupply || 0);

        const effectiveNextRewardValue = nextRewardValue.add(stakingRebase.mul(WBASHBalance).mul((index ?? BigNumber.from(0)).toHexString()));
        const wrappedTokenEquivalent = WBASHBalance.mul(index?.toHexString() || 0);

        return {
            fiveDayRate: `${fiveDayRate && (fiveDayRate * 100).toFixed(4)}  %`,
            stakingRebasePercentage: `${stakingRebase.mul(100).toFixed(2)} %`,
            nextRewardValue: `${nextRewardValue.toFixed(2)} BASH`,
            effectiveNextRewardValue: effectiveNextRewardValue ? `effectiveNextRewardValue wsBASH` : null,
            wrappedTokenEquivalent: `${wrappedTokenEquivalent?.toString() || 0} sBASH`,
            optionalMetrics: WBASHBalance.gt(0),
        };
    },
);
