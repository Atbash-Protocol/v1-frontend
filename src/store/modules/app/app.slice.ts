import { createSlice } from '@reduxjs/toolkit';

import { isActionRejected } from 'store/utils/action';

import { initializeProviderContracts, getBlockchainData, getCoreMetrics, getStakingMetrics } from './app.thunks';
import { ContractEnum, MainSliceState } from './app.types';

// Define the initial state using that type
const initialState: MainSliceState = {
    contracts: {
        [ContractEnum.BASH_CONTRACT]: null,
        [ContractEnum.SBASH_CONTRACT]: null,
        [ContractEnum.DAI_CONTRACT]: null,
        [ContractEnum.BASH_DAI_LP_ADDRESS]: null,
        [ContractEnum.STAKING_CONTRACT]: null,
        [ContractEnum.STAKING_HELPER_ADDRESS]: null,
        [ContractEnum.WSBASH_ADDRESS]: null,
        [ContractEnum.ZAPIN_ADDRESS]: null,
        [ContractEnum.REDEEM_CONTRACT]: null,
    },
    contractsLoaded: false,

    metrics: {
        circSupply: null,
        rawCircSupply: null,
        totalSupply: null,
        reserves: null,
        loading: true,
    },
    staking: {
        epoch: null,
        secondsToNextEpoch: null,
        index: null,
        loading: true,
    },
    blockchain: {
        timestamp: null,
        currentBlock: null,
        loading: true,
    },
    errorEncountered: false,
    loading: true,
};

export const MainSlice = createSlice({
    name: 'app-main',
    initialState,
    reducers: {},
    extraReducers: builder => {
        builder
            .addCase(initializeProviderContracts.pending, state => {
                return { ...state, contractsLoaded: false };
            })
            .addCase(initializeProviderContracts.fulfilled, (state, action) => {
                return { ...state, contracts: action.payload, contractsLoaded: true };
            })
            .addCase(getBlockchainData.fulfilled, (state, action) => {
                return {
                    ...state,
                    blockchain: {
                        ...action.payload,
                        loading: false,
                    },
                };
            })
            .addCase(getCoreMetrics.fulfilled, (state, action) => {
                return {
                    ...state,
                    metrics: {
                        ...action.payload,
                        loading: false,
                    },
                };
            })
            .addCase(getStakingMetrics.fulfilled, (state, action) => {
                return {
                    ...state,
                    staking: {
                        ...action.payload,
                        loading: false,
                    },
                };
            })
            .addMatcher(isActionRejected, (state, action) => {
                if (action.type.startsWith('app/')) state.errorEncountered = true;
            });
    },
});

export const selectContracts = (state: MainSliceState) => state.contracts;

export default MainSlice.reducer;
