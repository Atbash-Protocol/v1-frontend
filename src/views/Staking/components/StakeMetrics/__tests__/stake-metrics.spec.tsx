import { setUncaughtExceptionCaptureCallback } from 'process';

import { configureStore } from '@reduxjs/toolkit';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';

import * as AppAccountModule from 'store/modules/account/account.selectors';
import * as AppSelectorModule from 'store/modules/app/app.selectors';

import UserStakeMetrics from '..';

function renderComponent(component: JSX.Element) {
    return render(<Provider store={configureStore({ reducer: jest.fn() })}>{component}</Provider>);
}

describe('UserStakeMetrics', () => {
    const balances = {
        BASH: '1 BASH',
        SBASH: '0 SBASH',
        WSBASH: '1 WBASH',
    };
    const stakingInfos = {
        wrappedTokenEquivalent: '0 sBASH',
        nextRewardValue: '0 BASH',
        stakingRebasePercentage: '0.01 %',
        fiveDayRate: '',
        optionalMetrics: false,
        effectiveNextRewardValue: '0.1 wSBASH',
    };

    it('Renders', () => {
        jest.spyOn(AppAccountModule, 'selectFormattedStakeBalance').mockReturnValue({ balances } as any);
        jest.spyOn(AppSelectorModule, 'selectUserStakingInfos').mockReturnValue(stakingInfos);

        const component = renderComponent(<UserStakeMetrics />);

        expect(component).toMatchSnapshot();
    });

    it('Renders with optional metrics', () => {
        jest.spyOn(AppAccountModule, 'selectFormattedStakeBalance').mockReturnValue({ balances } as any);
        jest.spyOn(AppSelectorModule, 'selectUserStakingInfos').mockReturnValue({ ...stakingInfos, optionalMetrics: true });

        const component = renderComponent(<UserStakeMetrics />);

        expect(component).toMatchSnapshot();
    });
});
