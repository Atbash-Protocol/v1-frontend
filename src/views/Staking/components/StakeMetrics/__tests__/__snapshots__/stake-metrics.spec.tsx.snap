// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserStakeMetrics Renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-0"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-10zuisz-MuiTypography-root"
        >
          stake:StakingMetrics
           
        </h4>
        <div
          class="MuiBox-root css-1z137pa"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1qha2kt-MuiTypography-root"
          >
            YourBalance
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-le79b4-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-15tfpcm-MuiSkeleton-root"
            />
          </p>
        </div>
        <div
          class="MuiBox-root css-1z137pa"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1qha2kt-MuiTypography-root"
          >
            stake:YourStakedBalance
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-le79b4-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-15tfpcm-MuiSkeleton-root"
            />
          </p>
        </div>
        <div
          class="MuiBox-root css-1z137pa"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1qha2kt-MuiTypography-root"
          >
            stake:NextRewardAmount
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-le79b4-MuiTypography-root"
          >
            0 BASH
          </p>
        </div>
        <div
          class="MuiBox-root css-1z137pa"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1qha2kt-MuiTypography-root"
          >
            stake:NextRewardYield
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-le79b4-MuiTypography-root"
          >
            0.01 %
          </p>
        </div>
        <div
          class="MuiBox-root css-1z137pa"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1qha2kt-MuiTypography-root"
          >
            stake:ROIFiveDayRate
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-le79b4-MuiTypography-root"
          >
            
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-0"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-10zuisz-MuiTypography-root"
      >
        stake:StakingMetrics
         
      </h4>
      <div
        class="MuiBox-root css-1z137pa"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1qha2kt-MuiTypography-root"
        >
          YourBalance
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-le79b4-MuiTypography-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-15tfpcm-MuiSkeleton-root"
          />
        </p>
      </div>
      <div
        class="MuiBox-root css-1z137pa"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1qha2kt-MuiTypography-root"
        >
          stake:YourStakedBalance
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-le79b4-MuiTypography-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-15tfpcm-MuiSkeleton-root"
          />
        </p>
      </div>
      <div
        class="MuiBox-root css-1z137pa"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1qha2kt-MuiTypography-root"
        >
          stake:NextRewardAmount
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-le79b4-MuiTypography-root"
        >
          0 BASH
        </p>
      </div>
      <div
        class="MuiBox-root css-1z137pa"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1qha2kt-MuiTypography-root"
        >
          stake:NextRewardYield
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-le79b4-MuiTypography-root"
        >
          0.01 %
        </p>
      </div>
      <div
        class="MuiBox-root css-1z137pa"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1qha2kt-MuiTypography-root"
        >
          stake:ROIFiveDayRate
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-le79b4-MuiTypography-root"
        >
          
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UserStakeMetrics Renders with optional metrics 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-0"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-10zuisz-MuiTypography-root"
        >
          stake:StakingMetrics
           
        </h4>
        <div
          class="MuiBox-root css-1z137pa"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1qha2kt-MuiTypography-root"
          >
            YourBalance
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-le79b4-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-15tfpcm-MuiSkeleton-root"
            />
          </p>
        </div>
        <div
          class="MuiBox-root css-1z137pa"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1qha2kt-MuiTypography-root"
          >
            stake:YourStakedBalance
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-le79b4-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-15tfpcm-MuiSkeleton-root"
            />
          </p>
        </div>
        <div
          class="MuiBox-root css-1z137pa"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1qha2kt-MuiTypography-root"
          >
            stake:NextRewardAmount
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-le79b4-MuiTypography-root"
          >
            0 BASH
          </p>
        </div>
        <div
          class="MuiBox-root css-1z137pa"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1qha2kt-MuiTypography-root"
          >
            stake:NextRewardYield
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-le79b4-MuiTypography-root"
          >
            0.01 %
          </p>
        </div>
        <div
          class="MuiBox-root css-1z137pa"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1qha2kt-MuiTypography-root"
          >
            stake:ROIFiveDayRate
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-le79b4-MuiTypography-root"
          >
            
          </p>
        </div>
        <div
          class="MuiBox-root css-1z137pa"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1qha2kt-MuiTypography-root"
          >
            stake:EffectiveNextRewardAmount
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-le79b4-MuiTypography-root"
          >
            0.1 wSBASH
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-0"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-10zuisz-MuiTypography-root"
      >
        stake:StakingMetrics
         
      </h4>
      <div
        class="MuiBox-root css-1z137pa"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1qha2kt-MuiTypography-root"
        >
          YourBalance
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-le79b4-MuiTypography-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-15tfpcm-MuiSkeleton-root"
          />
        </p>
      </div>
      <div
        class="MuiBox-root css-1z137pa"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1qha2kt-MuiTypography-root"
        >
          stake:YourStakedBalance
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-le79b4-MuiTypography-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-15tfpcm-MuiSkeleton-root"
          />
        </p>
      </div>
      <div
        class="MuiBox-root css-1z137pa"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1qha2kt-MuiTypography-root"
        >
          stake:NextRewardAmount
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-le79b4-MuiTypography-root"
        >
          0 BASH
        </p>
      </div>
      <div
        class="MuiBox-root css-1z137pa"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1qha2kt-MuiTypography-root"
        >
          stake:NextRewardYield
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-le79b4-MuiTypography-root"
        >
          0.01 %
        </p>
      </div>
      <div
        class="MuiBox-root css-1z137pa"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1qha2kt-MuiTypography-root"
        >
          stake:ROIFiveDayRate
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-le79b4-MuiTypography-root"
        >
          
        </p>
      </div>
      <div
        class="MuiBox-root css-1z137pa"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1qha2kt-MuiTypography-root"
        >
          stake:EffectiveNextRewardAmount
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-le79b4-MuiTypography-root"
        >
          0.1 wSBASH
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
